### DISCUSSION TITLE
New `osxphotos import` command

### DISCUSSION THREAD

**Discussion by RhetTbull** [RhetTbull/osxphotos#764] [Show and tell]
I've added a new `import` command to osxphotos in [v0.51.0](https://github.com/RhetTbull/osxphotos/releases/tag/v0.51.0) so it can be used to import photos and videos into Photos.  It uses the same template system that `export` uses so you can use a template to specify the album(s), the metadata, etc. 

For example:

Import files into Photos and add to album named for month of the year in folder named for the 4-digit year of
the file creation date:

`osxphotos import /Volumes/photos/*.jpg --album "{created.year}/{created.month}" --split-folder "/"`

Import files and use the parent folder as the album:

`osxphotos import /Volumes/photos/Madrid/*.jpg --album "{filepath.parent.name}"`

adds all photos to the album "Madrid" (the name of the file's parent folder).

import photos and use exiftool to set metadata from the imported file

`osxphotos import /Volumes/photos/* --exiftool`

Import all photos in `/Volumes/Photos` and all subdirectories, use the filepath of the imported photo to automatically create folders and albums (but exclude the parent folder of `/Volumes/Photos` from the imported filepath:

`osxphotos import /Volumes/Photos/* --walk --album "{filepath}" --relative-to "/Volumes/Photos" --split-folders "/"`

Use `osxphotos help import`

**Comment by dalisoft** [comment]
@RhetTbull Thank you for nice feature, but for it's does not work, i don't know why.

There log

```
/Volumes/Photos 
❯ osxphotos import "$PWD/Photos/*" \
      --exiftool \
      --dup-check \
      --merge-keywords
Importing 0 files ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% -:--:--
Done: imported 0 files, 0 errors, 
/Volumes/Photos 
❯ ls Photos | wc -l
   10217
/Volumes/Photos 
❯ 
```

**Reply by RhetTbull** [reply]
@dalisoft your command line is wrong. Wildcards do not get expanded inside quotes though environment variables do so `"$PWD/Photos/*"` becomes `"/Volumes/Photos/*"` and osxphotos finds no photos with the literal name `*`.   Try the command without the quotes:

```sh
osxphotos import $PWD/Photos/* \
      --exiftool \
      --dup-check \
      --merge-keywords
```

**Reply by dalisoft** [reply]
@RhetTbull Thank you, yes it worked. Before such scripts worked but this time it's something new for me (bash scripting)
